source/Extractors/src/ExtractorFCCSW.cc

```diff
@@ -119,16 +139,79 @@ bool ExtractorFCCSW::analyze()
   xmlDetEcapReadout->InsertEndChild(xmlDetEcapSeg);
 
   auto xmlDetEcapId = m_xmlDoc->NewElement("id");
-  xmlDetEcapId->SetText("system:4,posneg:1,disc:5,module:16,component:4,x:-15,z:-15");
+  xmlDetEcapId->SetText("system:5,disc:5,module:16,component:4,x:-15,z:-15");
   xmlDetEcapReadout->InsertEndChild(xmlDetEcapId);
 
 
   // Initialize detector id
   short detId = c_defaultTrackerId;
 
+  auto xmlIncludeDefinitions = m_xmlDoc->NewElement("include");
+  xmlIncludeDefinitions->SetAttribute("ref", xmlDefinitionsDocName.c_str());
+  m_xmlNodeRoot->InsertEndChild(xmlIncludeDefinitions);
+  
+  auto xmlSubdetectorAssemblies = m_xmlDoc->NewElement("detectors");
+  m_xmlNodeRoot->InsertEndChild(xmlSubdetectorAssemblies);
+
+  auto xmlSubdetectorBeampipe = m_xmlDoc->NewElement("detector");
+  xmlSubdetectorBeampipe->SetAttribute("id", 0);
+  xmlSubdetectorBeampipe->SetAttribute("name", "beampipe");
+  xmlSubdetectorBeampipe->SetAttribute("type", "BeamTube");
+  auto xmlBeamPipeDimensions = m_xmlDoc->NewElement("dimensions");
+  xmlBeamPipeDimensions->SetAttribute("rmin", printWithUnit(0., c_precision, "mm").c_str());
+  xmlBeamPipeDimensions->SetAttribute("rmax", printWithUnit(15., c_precision, "mm").c_str());
+  xmlBeamPipeDimensions->SetAttribute("z", printWithUnit(500., c_precision, "mm").c_str());
```

Comment:
Why are the beam-pipe r_min, r_max set to r_min=0, r_max=15mm? I'd naturally expect r_min=20, r_max=21mm if this parameters are needed...